### ISSUE TITLE
Crash when removeAll in a MutableObservableArray2D

### ISSUE THREAD

**Issue by omerCoosto** [DeclarativeHub/Bond#626]
I'm using swift 4.2 with Bond version 7.3.3:

similar to #584, when i tried to replace my collection by removing and inserting new collection. Bond diff signal are incorrect.
The following method **setDatasource** is how i call the **removeAllItemsAndSections** (do the same with **removeAll**):

```
var datas = MutableObservableArray2D(Array2D())

    func setDataSource(_ newItems: Array2D<T, U>?) {
        self.datas.removeAllItemsAndSections()

        guard let newItems = newItems, newItems.sections.isEmpty == false else { return }
        self.datas.batchUpdate { tree in
            for node in newItems.children {
                tree.append(node)
            }
        }
    }
```
When initially inserting data everything works fine:
![insert](https://user-images.githubusercontent.com/34768792/56356842-3dda6b00-61da-11e9-97e6-81f5d54acb4f.png)

Then when replacing the data with the previous method, the sections to delete are incorrect:
![delete](https://user-images.githubusercontent.com/34768792/56356841-3dda6b00-61da-11e9-8e10-229adf67b45c.png)

**Comment by srdanrasic** [comment]
Looks like you've found a bug. Thanks for reporting this! I've released a fix in the version 7.4.3.

I would recommend you update to Xcode 10.1. It's compatible with Swift 4.2. Then you can also update to latest Bond version. If you prefer no to do that, here is a fixed method that you can put in your project and use that instead until you update:

```swift
extension MutableChangesetContainerProtocol where Changeset: TreeChangesetProtocol, Changeset.Collection: RangeReplaceableTreeProtocol {

    public func _removeAll() {
        descriptiveUpdate { (collection) -> [Operation] in
            let deletes = collection.children.indices.reversed().map { Operation.delete(at: [$0]) }
            collection.removeAll()
            return deletes
        }
    }
}
```

**Comment by omerCoosto** [comment]
Great, thanks! for now i'll use the extension code you wrote. 
I'm already using Xcode 10.2, but when i do the switch to swift 5 i'll update bond to the latest version.
